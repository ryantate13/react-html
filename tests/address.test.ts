import {address} from '../src';

describe('address', () => {
    it('renders with children', () => {
        expect(address({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(address({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(address()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(address({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(address({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(address({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(address({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(address({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(address({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(address({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(address({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(address({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(address({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(address({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(address({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(address({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(address({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(address({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(address({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(address({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(address({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(address({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(address({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(address({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(address({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(address({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(address({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(address({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(address({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(address({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(address({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(address({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(address({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(address({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(address({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(address({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(address({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(address({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(address({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(address({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(address({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(address({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(address({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(address({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(address({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(address({translate: "no"})).toMatchSnapshot();
    });
});