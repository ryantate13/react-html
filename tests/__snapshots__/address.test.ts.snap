// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address renders with accessKey="a,b,c,d" 1`] = `
<address
  accessKey="a,b,c,d"
/>
`;

exports[`address renders with autoCapitalize="characters" 1`] = `
<address
  autoCapitalize="characters"
/>
`;

exports[`address renders with autoCapitalize="none" 1`] = `
<address
  autoCapitalize="none"
/>
`;

exports[`address renders with autoCapitalize="off" 1`] = `
<address
  autoCapitalize="off"
/>
`;

exports[`address renders with autoCapitalize="on" 1`] = `
<address
  autoCapitalize="on"
/>
`;

exports[`address renders with autoCapitalize="sentences" 1`] = `
<address
  autoCapitalize="sentences"
/>
`;

exports[`address renders with autoCapitalize="words" 1`] = `
<address
  autoCapitalize="words"
/>
`;

exports[`address renders with children 1`] = `
<address>
  TEST
</address>
`;

exports[`address renders with className="foo bar baz" 1`] = `
<address
  className="foo bar baz"
/>
`;

exports[`address renders with contentEditable="false" 1`] = `
<address
  contentEditable="false"
/>
`;

exports[`address renders with contentEditable="true" 1`] = `
<address
  contentEditable="true"
/>
`;

exports[`address renders with dir="auto" 1`] = `
<address
  dir="auto"
/>
`;

exports[`address renders with dir="ltr" 1`] = `
<address
  dir="ltr"
/>
`;

exports[`address renders with dir="rtl" 1`] = `
<address
  dir="rtl"
/>
`;

exports[`address renders with draggable="false" 1`] = `
<address
  draggable="false"
/>
`;

exports[`address renders with draggable="true" 1`] = `
<address
  draggable="true"
/>
`;

exports[`address renders with empty attributes 1`] = `<address />`;

exports[`address renders with hidden="false" 1`] = `
<address
  hidden={false}
/>
`;

exports[`address renders with hidden="true" 1`] = `
<address
  hidden={true}
/>
`;

exports[`address renders with id="TEST" 1`] = `
<address
  id="TEST"
/>
`;

exports[`address renders with inputMode="decimal" 1`] = `
<address
  inputMode="decimal"
/>
`;

exports[`address renders with inputMode="email" 1`] = `
<address
  inputMode="email"
/>
`;

exports[`address renders with inputMode="none" 1`] = `
<address
  inputMode="none"
/>
`;

exports[`address renders with inputMode="numeric" 1`] = `
<address
  inputMode="numeric"
/>
`;

exports[`address renders with inputMode="search" 1`] = `
<address
  inputMode="search"
/>
`;

exports[`address renders with inputMode="tel" 1`] = `
<address
  inputMode="tel"
/>
`;

exports[`address renders with inputMode="text" 1`] = `
<address
  inputMode="text"
/>
`;

exports[`address renders with inputMode="url" 1`] = `
<address
  inputMode="url"
/>
`;

exports[`address renders with is="foo" 1`] = `
<address
  is="foo"
/>
`;

exports[`address renders with itemID="https://google.com/" 1`] = `
<address
  itemID="https://google.com/"
/>
`;

exports[`address renders with itemProp="https://google.com/" 1`] = `
<address
  itemProp="https://google.com/"
/>
`;

exports[`address renders with itemRef="foo bar" 1`] = `
<address
  itemRef="foo bar"
/>
`;

exports[`address renders with itemScope="false" 1`] = `
<address
  itemScope={false}
/>
`;

exports[`address renders with itemScope="true" 1`] = `
<address
  itemScope={true}
/>
`;

exports[`address renders with itemType="https://google.com/" 1`] = `
<address
  itemType="https://google.com/"
/>
`;

exports[`address renders with lang="" 1`] = `
<address
  lang=""
/>
`;

exports[`address renders with lang="es-419" 1`] = `
<address
  lang="es-419"
/>
`;

exports[`address renders with slot="TEST" 1`] = `
<address
  slot="TEST"
/>
`;

exports[`address renders with spellCheck="false" 1`] = `
<address
  spellCheck="false"
/>
`;

exports[`address renders with spellCheck="true" 1`] = `
<address
  spellCheck="true"
/>
`;

exports[`address renders with style="[object Object]" 1`] = `
<address
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`address renders with tabIndex="1" 1`] = `
<address
  tabIndex={1}
/>
`;

exports[`address renders with title="TEST" 1`] = `
<address
  title="TEST"
/>
`;

exports[`address renders with translate="no" 1`] = `
<address
  translate="no"
/>
`;

exports[`address renders with translate="yes" 1`] = `
<address
  translate="yes"
/>
`;

exports[`address renders without attributes 1`] = `<address />`;
